src/Registration/cReg/cReg.cpp

```diff
@@ -447,13 +447,16 @@ void* cReg_NiftiImageData_get_inner_product(const void* im1_ptr, const void* im2
     CATCH;
 }
 extern "C"
-void* cReg_NiftiImageData_from_SIRFImageData(void* ptr)
+void* cReg_NiftiImageData_from_SIRFImageData(void* ptr, const int is_3D)
 {
-	try {
+    try {
         ImageData& sirf_im = objectFromHandle<ImageData>(ptr);
-        std::shared_ptr<NiftiImageData<float> >
-            sptr(new NiftiImageData<float>(sirf_im));
-        return newObjectHandle(sptr);
+
+        // Use an int as if bool to optionally run 3D checks
+        if (is_3D == 1)
```

Comment:
```suggestion
        if (is_3D)
```

Reply:
Can be, I put it in to be explicit as I was using `int` instead of `bool`. Feel free to commit the suggestion.